### PR TITLE
Avijit/build script changes

### PR DESCRIPTION
**Pull request by avijit-nervana**
* Refactored the build_ngtf.py by creating tools/build_utils.py 
* Added an option "--artifact_dir" to allow specifying the artifacts directory

### CODE REVIEW

File: tools/test_inference_models.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        raise "Need to provide artifacts directory"
+
+    if not (arguments.python_location):
+        raise "Need to provide the location of Python directory"
+
+    if not (arguments.model_location):
+        raise "Need to provide the location of models directory"
+
+    # Check to make sure that there is TensorFlow installed and will work
+    # TODO
```
Comment (sayantan-nervana) [inline tools/test_inference_models.py]: Do you mean something like this in the TODO? 
```
import tensorflow as tf
assert (tf.__version__ == '1.13.1')
```

Comment (sayantan-nervana) [reply tools/test_inference_models.py]: Should we put `test_inference_models.py` under `tools` or `test`?

Comment (avijit-nervana) [reply tools/test_inference_models.py]: No. We need to find out what version of TF was used to build NG and what version of TF this one is.

Comment (avijit-nervana) [reply tools/test_inference_models.py]: @sayantan-nervana 
>Should we put test_inference_models.py under tools or test

No. I am planning to move these to a new ci directory. for now this is temp.

**Pull request merged**